modules/apps/site/site-impl/src/main/java/com/liferay/site/internal/configuration/manager/MenuAccessConfigurationManagerImpl.java

```diff
@@ -57,26 +62,34 @@ public void addAccessRoleToControlMenu(Role role) throws Exception {
 
 	@Override
 	public void deleteRoleAccessToControlMenu(Role role) throws Exception {
-		for (Group group :
-				_groupLocalService.getGroups(
-					role.getCompanyId(), GroupConstants.ANY_PARENT_GROUP_ID,
-					true)) {
+		String roleId = String.valueOf(role.getRoleId());
 
-			MenuAccessConfiguration menuAccessConfiguration =
-				_configurationProvider.getGroupConfiguration(
-					MenuAccessConfiguration.class, group.getGroupId());
+		String filterString = StringBundler.concat(
+			"(&(service.factoryPid=", MenuAccessConfiguration.class.getName(),
+			".scoped))");
 
-			String roleId = String.valueOf(role.getRoleId());
-			String[] accessToControlMenuRoleIds =
-				menuAccessConfiguration.accessToControlMenuRoleIds();
+		Configuration[] configurations = _configurationAdmin.listConfigurations(
+			filterString);
+
+		if (configurations == null) {
+			return;
+		}
+		
+		for (Configuration configuration : configurations) {
+
+			Dictionary<String, Object> properties =
+				configuration.getProperties();
+
+			String[] accessToControlMenuRoleIds = (String[])properties.get(
+				"accessToControlMenuRoleIds");
 
 			if (ArrayUtil.contains(accessToControlMenuRoleIds, roleId)) {
 				accessToControlMenuRoleIds = ArrayUtil.remove(
 					accessToControlMenuRoleIds, roleId);
 
 				updateMenuAccessConfiguration(
-					group.getGroupId(), accessToControlMenuRoleIds,
-					menuAccessConfiguration.showControlMenuByRole());
+					(long)properties.get("groupId"), accessToControlMenuRoleIds,
+					(boolean)properties.get("showControlMenuByRole"));
```

Comment:
Since performance is a concern, and you have the configuration already here, I would probably just use `configuration.update(properties)` instead of `updateMenuAccessConfiguration` here to avoid re-access.